Code/GraphMol/MolEnumerator/StereoIsomer.cpp

```diff
@@ -0,0 +1,295 @@
+#include "MolEnumerator.h"
+#include <RDGeneral/Exceptions.h>
+
+#include <boost/range/irange.hpp>
+#include <unordered_set>
+#include <GraphMol/MolEnumerator/MolEnumerator.h>
+#include <GraphMol/Chirality.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/DistGeomHelpers/Embedder.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+
+namespace RDKit {
+namespace MolEnumerator {
+
+inline namespace detail {
+class StereoFlipper {
+ public:
+  StereoFlipper() = default;
+  virtual ~StereoFlipper() = default;
+
+  virtual void flip(RWMol& mol, bool flag) = 0;
+};
+}  // namespace detail
+
+namespace {
+
+class AtomStereoFlipper : public StereoFlipper {
+ public:
+  AtomStereoFlipper(Atom* atom) : d_atom(atom->getIdx()) {}
+
+  void flip(RWMol& mol, bool flag) final {
+    auto next_atom_stereo = flag ? Atom::ChiralType::CHI_TETRAHEDRAL_CW
+                                 : Atom::ChiralType::CHI_TETRAHEDRAL_CCW;
+    mol.getAtomWithIdx(d_atom)->setChiralTag(next_atom_stereo);
+  }
+  static void add_flippers_from_mol(const ROMol& mol,
+                                    const StereoEnumerationOptions& options,
+                                    std::vector<stereo_flipper_t>& flippers) {
+    if (options.onlyStereoGroups) {
+      return;
+    }
+
+    for (auto* atom : mol.atoms()) {
```

Comment:
What's the difference between:

    for (auto* atom: mol.atoms()) {
     
and

    for (auto atom: mol.atoms()) {

given that atom is a pointer?

Reply:
in this case there's no difference, it's just for readability. there's only a difference for `const auto` vs `const auto*` https://stackoverflow.com/questions/36563017/difference-between-const-auto-and-const-auto